### PR TITLE
[Caching] Use subInvocation to verify interface

### PR DESCRIPTION
**Pull request by cachemeifyoucan**
The interface verification needs to read some command-line from the interface file so it needs to be built inside sub-invocation.

rdar://122423965

### CODE REVIEW

**Comment by cachemeifyoucan** [comment]
@swift-ci please smoke test

**Comment by cachemeifyoucan** [comment]
Please test with following pull request:
https://github.com/apple/llvm-project/pull/8165

@swift-ci please smoke test

**Comment by cachemeifyoucan** [comment]
Please test with following pull request:
https://github.com/apple/llvm-project/pull/8165

@swift-ci please smoke test

File: lib/AST/ASTContext.cpp [hunk 1]
```diff
@@ -722,17 +721,17 @@ ASTContext::ASTContext(
     The##SHORT_ID##Type(new (*this, AllocationArena::Permanent) \
                           ID##Type(*this)),
 #include "swift/AST/TypeNodes.def"
-      TheIEEE32Type(new (*this, AllocationArena::Permanent)
+      TheIEEE32Type(new(*this, AllocationArena::Permanent)
```
Comment (artemcm) [inline lib/AST/ASTContext.cpp:724]: Unrelated change, I think.

**Comment by cachemeifyoucan** [comment]
Please test with following pull request:
https://github.com/apple/llvm-project/pull/8165

@swift-ci please smoke test

**Comment by cachemeifyoucan** [comment]
Please test with following pull request:
https://github.com/apple/llvm-project/pull/8165

@swift-ci please smoke test

**Comment by cachemeifyoucan** [comment]
Please test with following pull request:
https://github.com/apple/llvm-project/pull/8165

@swift-ci please smoke test macOS platform

**Comment by cachemeifyoucan** [comment]
Please test with following pull request:
https://github.com/apple/llvm-project/pull/8165

@swift-ci please smoke test macOS platform

**Comment by cachemeifyoucan** [comment]
Please test with following pull request:
https://github.com/apple/llvm-project/pull/8165

@swift-ci please smoke test

**Comment by cachemeifyoucan** [comment]
Please test with following pull request:
https://github.com/apple/llvm-project/pull/8165

@swift-ci please smoke test

**Comment by cachemeifyoucan** [comment]
ping

**Review by artemcm — APPROVED** [review]
Looks great.
